tokio/src/runtime/basic_scheduler.rs

```diff
@@ -539,7 +585,15 @@ impl CoreGuard<'_> {
                 // pending I/O events.
                 core = context.park_yield(core);
             }
-        })
+        });
+
+        match ret {
+            Some(ret) => ret,
+            None => {
+                // `block_on` panicked.
+                panic!("a spawned task panicked and the runtime is configured to shutdown on unhandled panic");
```

Comment:
it's kind of a bummer that we just create a new panic here, rather than continuing the original panic...

Reply:
I agree, but I don't think its a big deal. The panic is still printed to stderr by the panic handler.